stablehlo/reference/ProcessGrid.cpp

```diff
@@ -162,6 +197,21 @@ ProcessGroups ProcessGrid::flattenedIds(
 
 void ProcessGrid::outfeed(ArrayRef<Tensor> inputs) { outfeed_.push(inputs); }
 
+SmallVector<Tensor> ProcessGrid::recv(ChannelId channelId,
+                                      ProcessId processId) {
+  sendRecvReady_.insert(channelId);
+  sendRecvConditions_[channelId].notify_one();
+
+  auto &state = sendRecvChannels_[channelId];
+  std::unique_lock<std::mutex> lock(state.mutex);
```

Comment:
send uses:

```
std::unique_lock<std::mutex> lock(sendRecvChannels_[channelId].mutex);
```

Maybe pick one representation and use it both places for consistency